tests/utils/test_fslock.py

```diff
@@ -0,0 +1,62 @@
+import os
+import time
+from threading import Thread
+
+from fsspec.implementations.local import LocalFileSystem
+
+from mlem.utils.fslock import LOCK_EXT, FSLock
+from mlem.utils.path import make_posix
+
+NAME = "testlock"
+
+
+# pylint: disable=protected-access
+def test_fslock(tmpdir):
+    fs = LocalFileSystem()
+    lock = FSLock(fs, tmpdir, NAME)
+
+    with lock:
+        assert lock._timestamp is not None
+        assert lock._salt is not None
+        lock_path = make_posix(
+            os.path.join(
+                tmpdir, f"{NAME}.{lock._timestamp}.{lock._salt}.{LOCK_EXT}"
+            )
+        )
+        assert lock.lock_path == lock_path
+        assert fs.exists(lock_path)
+
+    assert lock._timestamp is None
+    assert lock._salt is None
+    assert not fs.exists(lock_path)
+
+
+def _work(dirname, num):
+    time.sleep(0.3 + num / 5)
```

Comment:
`0.3 + num / 5` ??

Reply:
it so that all _work calls ran a different time